endpoints/users/controller.py

```diff
@@ -35,43 +39,45 @@
 
 
 class ListUsersResource(Resource):
+    @jwt_required
     def get(self):
         try:
-            users = User.query.all()
-            users = [marshal(user, user_list_fields) for user in users]
-            return users
-
-        except Exception as error:
-            response.errorResponse(str(error))
-            return response.__dict__
+            users = userRepository.find()
+            return marshal(users, user_list_fields)
+        except:
+            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
 
 
 class RegisterUserResource(Resource):
     def post(self):
         try:
-            user = register_parser.parse_args()
+            credentials = register_parser.parse_args()
+            user = userRepository.findByUsername(credentials['username'])
+            if(bool(user)):
+                return {'message': 'User Already Registered', 'timestamp': round(time.time())}, 500
             hashedPass = bcrypt.hashpw(
-                user['password'].encode('utf-8'), bcrypt.gensalt())
-            user['password'] = hashedPass.decode('utf-8')
-            db.session.add(User(**user))
-            db.session.commit()
-            return marshal(user, user_list_fields)
+                credentials['password'].encode('utf-8'), bcrypt.gensalt())
+            credentials['password'] = hashedPass.decode('utf-8')
+            userRepository.add(credentials)
+            return marshal(credentials, user_list_fields)
 
-        except Exception as error:
-            response.errorResponse(str(error))
-            return response.__dict__
+        except:
+            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
 
 
 class UsersByIdResource(Resource):
+    @jwt_required
     def get(self, id=None):
-        user = User.query.filter_by(id=id).first()
-        return marshal(user, user_list_fields)
+        try:
+            user = userRepository.findById(id)
+            return marshal(user, user_list_fields)
+        except:
+            return {'message': 'Something went wrong', 'timestamp': round(time.time())}, 500
 
+    @jwt_required
     def delete(self, id=None):
-        user = User.query.get(id)
-        db.session.delete(user)
-        db.session.commit()
-        return marshal(user, user_list_fields)
+        userRepository.delete(id)
```

Comment:
Probably should check that the user that is trying to delete is either an admin or the current user